src/lib/components/Modal/ModalHead.jsx

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  withGlobalProps,
+} from '../../provider';
+import { classNames } from '../../utils/classNames';
+import styles from './ModalHead.scss';
+
+export const ModalHead = ({
+  children,
+  id,
+  justify,
+}) => {
+  const justifyClass = (value) => {
+    if (value === 'center') {
+      return styles.isJustifiedToCenter;
+    }
+
+    if (value === 'space-between') {
+      return styles.isJustifiedToSpaceBetween;
+    }
+
+    return styles.isJustifiedToStretch;
+  };
+
+  return (
+    <div
+      className={classNames(
+        styles.root,
+        justifyClass(justify),
+      )}
+      id={id}
+    >
+      {children}
+    </div>
+  );
+};
+
+ModalHead.defaultProps = {
+  id: undefined,
+  justify: 'space-between',
+};
+
+ModalHead.propTypes = {
+  /**
+   * Content of the head (preferably heading element with ModalCloseButton element).
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * ID of the root HTML element.
+   */
+  id: PropTypes.string,
+  /**
+   * Horizontal alignment (distribution) of individual buttons.
+   */
+  justify: PropTypes.oneOf(['center', 'space-between', 'stretch']),
```

Comment:
@adamkudrna Do you think we need it? The way I look at this is that even examples shown in the docs are artificially created examples just to demonstrate the possibility and I am not so happy with that. Besides that `space-between` can simulate `left`, and  `right` is not IMHO necessary, maybe in RTL environment?

So you are really sure we want to open it up like this? I am afraid of creating non-sense layouts because of those options (e.g. I have never seen title on the right and I would not expect that in LTR environments and I frankly do not know whether RTL just switches direction of text or whether the alignment changes to the right as well.)

Reply:
I think the options should be the same as in `ModalFooter` as @adamkudrna suggests because:

1. I can envision header layout such as: 
    ```
    X | My Modal Title
    ```
2. having the same options leads to less developer confusion and less need to consult the docs
3. Code can be shared <-- I agree this is a small benefit


Note: In order to support RTL we should not use `left` / `right`, but we should use `start` / `center` (as we already do in `ModalFooter`).

Reply:
Not happy with that, but ok, lets have it same.

Reply:
@mbohal Thanks for a nice summary. 👍🏻 

Although I'm not a fan of it, Apple prefers putting the close button on the left:

<img width="1182" alt="obrazek" src="https://user-images.githubusercontent.com/5614085/182187440-36c6910b-33df-4a92-bc66-a82f55b8d0d6.png">